scripts/docker/clean-logs.sh

```diff
@@ -21,19 +21,30 @@ set -euo pipefail
 
 readonly DIRECTORY="${AIRFLOW_HOME:-/usr/local/airflow}"
 readonly RETENTION="${AIRFLOW__LOG_RETENTION_DAYS:-15}"
+readonly DELETE_LOGS_FOLDERS="${AIRFLOW__LOG_DELETE_FOLDERS:-true}"
 
 trap "exit" INT TERM
 
 readonly EVERY=$((15*60))
 
+echo "Running the improved cleaning logs..."
 echo "Cleaning logs every $EVERY seconds"
 
 while true; do
-  echo "Trimming airflow logs to ${RETENTION} days."
-  find "${DIRECTORY}"/logs \
-    -type d -name 'lost+found' -prune -o \
-    -type f -mtime +"${RETENTION}" -name '*.log' -print0 | \
-    xargs -0 rm -f
+  if [[ "${DELETE_LOGS_FOLDERS}" == "true" ]]
+  then
+    echo "Trimming airflow logs folders to ${RETENTION} days."
+    find "${DIRECTORY}"/logs \
+      -type d -name 'lost+found' -prune -o \
+      -type d -mtime +"${RETENTION}" -print0 | \
```

Comment:
This won't work as written. mtime updates don't propogate up, so you'd end up deleting all of a dags logs after the root folder has been around for $retention_days.

Instead, we could have a second command that goes through and deletes old empty folders? I think the "empty" aspect is critical.

Reply:
Good point @jedcunningham  adding `-empty` predicate in find should solve the issue.